docs/InstrumentationVerification/InstrumentationVerification.md

```diff
@@ -0,0 +1,134 @@
+    2. By the Debugger team, to verify that adding probes produces valid IL on a wide range of methods
+2. As a diagnostic tool, the InstrumentationVerification tool can be used to quickly triage support cases where the customer is complaining that our CLR Profiler causes a `BadImageFormationException`, `InvalidProgramException`, `TypeLoadException`, etc etc. 
+
+    A primary goal of this tool is to support the scenario where we need to perform a post-mortem diagnosis of a crash that we suspect may have been caused due to faulty instrumentation (for example, a crash due to an ExecutionEngineException or an AccessViolationException). 
+
+3. As part of development of new bytecode instrumentation techniques, enabling breakpoint debugging of instrumented methods via [dnSpy](https://github.com/dnSpy/dnSpy).  
+
+
+# Usage
+
+To run a verification, simply:
+
+
+
+1. Add the environment variable `DD_TRACE_WRITE_INSTRUMENTATION_TO_DISK`, set to “true”.
+2. Run or restart the application we are instrumenting. Observe that as we instrument the app, more and more files will appear under [Datadog Logs Folder]/InstrumentationVerification/{ProcessName}_{ProcessID}_{ProcessCreationTime}
+3. Via command line, run the command: 
+
+    Datadog.InstrumentedAssemblyVerification.Standalone.exe [Full path of the folder generated in step 2] 
+
+
+    Once the command has finished executing, you will find new files under the InstrumentationVerification folder:
+
+    1. Logs file of the verification tool for each assembly will appear under a folder named **OUPUT_InstrumentedAssemblies/InstrumentationVerificationLogger_[DATE]_.log**)
+    2. The instrumented assemblies will appear under a folder named **OUPUT_InstrumentedAssemblies**
+
+**Note:** When handling a support case, the first two steps must be performed on the customer’s site. However, if the customer is willing to upload and send us the instrumentation output folder (which is conveniently placed inside the Datadog Logs folder), then the 3rd step can be performed offline.
+
+
+# Approach & High Level Design
+
+
+![alt_text](InstrumentationVerificationDesign.png "instrumentation verification high level design")
+
+
+The Instrumentation Verification process is divided into three main steps:
+
+
+
+1. At runtime, we write the following information to disk:
+    1. A copy of every assembly that was loaded 
+    2. Several “metadata” files which contains all the metadata changes that were performed via _IMetadataEmit2 / IMetadataAssemblyEmit._ The file format is textual, in order to assist with easy troubleshooting.
+    3. A “bytecode” file, which contains all the IL bytecode that was modified via _ICorProfilerInfo::SetILFunctionBody_ or _ICorProfilerFunctionControl::SetILFunctionBody,_ in binary format.
+
+    
+
+    The files in section (a) are written to a folder named **INPUT_OriginalAssemblies** whereas the files in sections (b) and (c) are written to a folder named **INPUT_InstrumentationLog.**
```

Comment:
Ignore me if the list rendering depends on the setup / browser
```suggestion
    The files in section (i) are written to a folder named **INPUT_OriginalAssemblies** whereas the files in sections (ii) and (iii) are written to a folder named **INPUT_InstrumentationLog.**
```

Reply:
Fixed in 4c7c44dcf9bc4f5affbde6c20b8fd17d1735d225